### PR TITLE
Initial StatsReporter for Source 

### PR DESCRIPTION
**Pull request by nachocano**
- Adding a `source_stats_reporter` within metrics package. As with the `source` duck type, eventing sources might not want to depend on eventing to construct a source/importer. Thus, we are adding a source reporter herein. Note the name is source instead of importer, to match the duck type. Label tags are importer though. We may need to change them later on..

### CODE REVIEW

**Comment by nachocano** [comment]
/assign @yanweiguo 
/assign @n3wscott 

fyi @sayanh, @Harwayne

File: metrics/source_stats_reporter.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	Namespace     string
+	EventType     string
+	EventSource   string
+	Name          string
+	ResourceGroup string
+}
+
+// StatsReporter defines the interface for sending source metrics.
+type StatsReporter interface {
+	ReportEventCount(args *ReportArgs, responseCode int) error
```
Comment (yanweiguo) [inline metrics/source_stats_reporter.go]: Could you add a comment for this func? Looks like it only reports one per call?

File: metrics/config_test.go [hunk 2]
```diff
@@ -1,9 +1,12 @@
 /*
-Copyright 2018 The Knative Authors.
+Copyright 2018 The Knative Authors
```
Comment (mattmoor-sockpuppet) [inline metrics/config_test.go]: Incorrect copyright year:
```suggestion
Copyright 2019 The Knative Authors
```

Comment (nachocano) [reply metrics/config_test.go]: Done

File: metrics/source_stats_reporter.go (hunk 1 above)
Comment (nachocano) [reply metrics/source_stats_reporter.go]: Yes, added a comment. Thanks!

File: logging/config.go [hunk 3]
```diff
@@ -196,3 +199,58 @@ func ConfigMapName() string {
 	}
 	return cm
 }
+
+// Base64ToLoggingConfig converts a json+base64 string of a Config.
+// Returns a non-nil Config always.
+func Base64ToLoggingConfig(base64 string) (*Config, error) {
```
[the remark this answers is not in the document]
Comment (nachocano) [reply logging/config.go]: I think we can do that. I'm fine with getting this in, and then do the change...

[the remark this answers is not in the document]
Comment (nachocano) [reply logging/config.go]: @Harwayne suggested the change, let's see what he thinks...

[the remark this answers is not in the document]
Comment (nachocano) [reply logging/config.go]: removed it, it was easier...

**Comment by n3wscott** [comment]
I would break this into at least 2 PRs.

File: metrics/config_test.go [hunk 4]
```diff
@@ -1,9 +1,12 @@
 /*
-Copyright 2018 The Knative Authors.
+Copyright 2019 The Knative Authors
```
Comment (n3wscott) [inline metrics/config_test.go]: this is not a new file so the CR should stay right?

Comment (nachocano) [reply metrics/config_test.go]: > I would break this into at least 2 PRs.

will do

Comment (nachocano) [reply metrics/config_test.go]: > this is not a new file so the CR should stay right?

yeah, sock-puppet suggestion, so I changed it, will roll it back

**Comment by nachocano** [comment]
@n3wscott  splitted in this one and https://github.com/knative/pkg/pull/656
ready for another look

**Comment by yanweiguo** [comment]
/lgtm

I'll approve if @n3wscott has no more comments.

**Comment by nachocano** [comment]
synced offline with @n3wscott and he is willing to accept this. Let's wait to hear from him here, and we are good to go... thanks @yanweiguo

**Pull request merged**